### PR TITLE
Improve build pipeline

### PR DESCRIPTION
**Pull request by bcheung**
Cleans up Makefile and runs the build container as non-root user.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-sudo: required
```
Comment (anfernee) [inline .travis.yml:1]: this is great!!

File: Makefile [hunk 2]
```diff
@@ -42,22 +42,35 @@ ALL_ARCH := amd64 arm arm64 ppc64le
...
 
 IMAGE := $(REGISTRY)/$(BIN)-$(ARCH)
 
 BUILD_IMAGE ?= golang:1.14-alpine
 
+# Docker run command prefix
+DOCKER_RUN =  docker run                                                            \
+				--rm                                                                \
+				-u $$(id -u):$$(id -g)                                              \
+				-v "$$(pwd)/.go/.cache:/.cache"                                     \
```
Comment (anfernee) [inline Makefile:65]: `$(CURDIR)`` as opposed to shelling out?

**Comment by anfernee** [comment]
Look great. Can you also add `.PHONY` for all the non-file targets?

**Comment by anfernee** [comment]
@lzang can you take a look?

File: .gitignore [hunk 3]
```diff
@@ -4,6 +4,10 @@
 # OSX trash
 .DS_Store
 
+# Intellij files
```
Comment (lzang) [inline .gitignore:7]: Why do you have these files?

Comment (bcheung) [reply .gitignore:7]: These are just files that Intellij Goland IDE generates. (like the Vscode files)

**Pull request merged**